src/models/state/mempool.rs

```diff
@@ -470,7 +529,7 @@ mod tests {
     use super::*;
 
     #[tokio::test]
-    pub async fn insert_then_get_then_remove_then_get() {
+    pub async fn insert_then_get_then_remove_then_get() -> Result<()> {
```

Comment:
If the test does not return anything, then we get a nice stack trace whenever it fails. If the test returns some error, then the stack trace is not as useful, is my experience. For this reason, I prefer using `unwrap` or `expect` in tests, not the `?` operator.

Reply:
backtraces can be automatically displayed for anyhow errors via:

cargo.toml
```
anyhow = { version = "1.0", features = ["backtrace"] }
```

.cargo/config.toml
```
[env]
RUST_BACKTRACE="1"
```

With these enabled, when a test returns an anyhow::Error a backtrace is displayed.   example:

```
---- rpc_server::rpc_server_tests::send_to_many_test stdout ----
Error: Do we get a stack trace?

Stack backtrace:
   0: anyhow::error::<impl anyhow::Error>::msg
             at /home/danda/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.81/src/backtrace.rs:27:14
   1: anyhow::__private::format_err
             at /home/danda/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.81/src/lib.rs:688:13
   2: neptune_core::rpc_server::rpc_server_tests::send_to_many_test::{{closure}}
             at ./src/rpc_server.rs:1399:9
<truncated>
```

I will add these changes to the PR. 

I choose to avoid unwrap/expect whenever possible in code I author, but I agree it is nice to have backtraces for errors in tests (and elsewhere).  Fortunately, its possible to do both.